harambe/parser/parser.py

```diff
@@ -159,6 +171,79 @@ def _schema_to_pydantic_model(
 
         return create_model(model_name, __config__=config, **fields)  # type: ignore
 
+    def _get_all_required_fields(
+        self, schema: Schema, parent_key: str = ""
+    ) -> List[str]:
+        """
+        Recursively collect all required fields from the schema, including nested ones.
+        """
+        required_fields = []
+        for field_name, field_info in schema.items():
+            full_key = f"{parent_key}.{field_name}" if parent_key else field_name
+            if field_info.get("required", False):
+                required_fields.append(full_key)
+
+            if field_info.get("type") == OBJECT_TYPE:
+                nested_properties = field_info.get("properties", {})
+                required_fields.extend(
+                    self._get_all_required_fields(nested_properties, full_key)
+                )
+            elif field_info.get("type") == LIST_TYPE:
+                item_info = field_info.get("items", {})
+                if item_info.get("type") == OBJECT_TYPE:
+                    nested_properties = item_info.get("properties", {})
+                    required_fields.extend(
+                        self._get_all_required_fields(nested_properties, full_key)
+                    )
+
+        return required_fields
```

Comment:
no sure why we need this? Since the schema is a dict of dicts of dicts access time should be O(1) to check if the field is required. IMHO i could be simpler to just do this below